components/script/dom/readablestreambyobreader.rs

```diff
@@ -194,9 +258,130 @@ impl ReadableStreamBYOBReaderMethods<crate::DomTypeHolder> for ReadableStreamBYO
     }
 
     /// <https://streams.spec.whatwg.org/#byob-reader-read>
-    fn Read(&self, _view: CustomAutoRooterGuard<ArrayBufferView>, can_gc: CanGc) -> Rc<Promise> {
-        // TODO
-        Promise::new(&self.reflector_.global(), can_gc)
+    #[allow(unsafe_code)]
+    fn Read(
+        &self,
+        view: CustomAutoRooterGuard<ArrayBufferView>,
+        options: &ReadableStreamBYOBReaderReadOptions,
+        can_gc: CanGc,
+    ) -> Rc<Promise> {
+        let view = HeapBufferSource::<ArrayBufferViewU8>::new(BufferSource::Uint8Array(
```

Comment:
Actually here I think there may be a problem, because the view could be either one of the type arrays or a data view. 
So, why `ArrayBufferViewU8`, and not `ArrayBufferView`? And also to get the right buffer source we would need to get the type of the array(or dataview)?

Reply:
Perhaps use `view.has_typed_array_name` and either set the buffer source to a data view or a `Uint8Array`?

Reply:
Or we reduce `BufferSource` to two variants, one `ArrayBuffer`, and one `ArrayBufferView`, and here we would use `ArrayBufferView`.

Reply:
> Or we reduce

Starting to think that is the right solution; we don't do anything with the more specific enum variants anyway.